src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchCaseSearchOperation.java

```diff
@@ -49,37 +55,60 @@ public ElasticsearchCaseSearchOperation(JestClient jestClient,
     }
 
     @Override
-    public CaseSearchResult execute(CaseSearchRequest caseSearchRequest) {
-        SearchResult result = search(caseSearchRequest);
+    public CaseSearchResult execute(CrossCaseTypeSearchRequest request) {
+        MultiSearchResult result = search(request);
         if (result.isSucceeded()) {
             return toCaseDetailsSearchResult(result);
         } else {
             throw new BadSearchRequest(result.getErrorMessage());
         }
     }
 
-    private SearchResult search(CaseSearchRequest caseSearchRequest) {
-        Search searchRequest = secureAndTransformSearchRequest(caseSearchRequest);
+    private MultiSearchResult search(CrossCaseTypeSearchRequest request) {
+        MultiSearch multiSearchAction = secureAndTransformSearchRequest(request);
         try {
-            return jestClient.execute(searchRequest);
+            return jestClient.execute(multiSearchAction);
         } catch (IOException e) {
             throw new ServiceException("Exception executing Elasticsearch : " + e.getMessage(), e);
         }
     }
 
-    private Search secureAndTransformSearchRequest(CaseSearchRequest caseSearchRequest) {
-        CaseSearchRequest securedSearchRequest = caseSearchRequestSecurity.createSecuredSearchRequest(caseSearchRequest);
+    private MultiSearch secureAndTransformSearchRequest(CrossCaseTypeSearchRequest request) {
+        Collection<Search> securedSearchActions = request.getCaseTypeIds()
+            .stream()
+            .map(caseTypeId -> createSecuredSearchAction(caseTypeId, request.getSearchRequestJsonNode()))
+            .collect(toList());
+
+        return new MultiSearch.Builder(securedSearchActions).build();
+    }
+
+    private Search createSecuredSearchAction(String caseTypeId, JsonNode searchRequestJsonNode) {
+        CaseSearchRequest securedSearchRequest = caseSearchRequestSecurity.createSecuredSearchRequest(new CaseSearchRequest(caseTypeId, searchRequestJsonNode));
         return new Search.Builder(securedSearchRequest.toJsonString())
-            .addIndex(getCaseIndexName(caseSearchRequest.getCaseTypeId()))
+            .addIndex(getCaseIndexName(caseTypeId))
             .addType(getCaseIndexType())
             .build();
     }
 
-    private CaseSearchResult toCaseDetailsSearchResult(SearchResult result) {
-        List<String> casesAsString = result.getSourceAsStringList();
+    private CaseSearchResult toCaseDetailsSearchResult(MultiSearchResult multiSearchResult) {
+        List<CaseDetails> caseDetails = new ArrayList<>();
+        AtomicLong totalHits = new AtomicLong(0);
```

Comment:
do we really need an AtomicLong here?

Reply:
not really, it was just because of lambda. Will use traditional for loop instead.